src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -1491,14 +1491,21 @@ def value_counts(
         apparitions of values, divide the index in the specified
         number of half-open bins.
         """
-        # TODO: SNOW-1458133 implement value_counts
-        WarningMessage.index_to_pandas_warning("value_counts")
-        return self.to_pandas().value_counts(
-            normalize=normalize,
-            sort=sort,
-            ascending=ascending,
-            bins=bins,
-            dropna=dropna,
+        if bins is not None:
+            ErrorMessage.not_implemented(
+                "`bins` is not yet supported for Index.value_counts."
```

Comment:
once you have a separate query_compiler interface for index, please move this check to the query compiler. Wether a parameter is currently supported or not is the responsibility of backend